go-controller/cmd/ovnkube-trace/ovnkube-trace.go

```diff
@@ -26,43 +27,78 @@ import (
 	util "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
 )
 
-type AddrInfo struct {
-	Family    string `json:"family,omitempty"`
-	Local     string `json:"local,omitempty"`
-	Prefixlen int    `json:"prefixlen,omitempty"`
-}
-type IpAddrReq struct {
-	IfIndex   int        `json:"ifindex,omitempty"`
-	IfName    string     `json:"ifname,omitempty"`
-	LinkIndex int        `json:"link_index,omitempty"`
-	AInfo     []AddrInfo `json:"addr_info,omitempty"`
-}
+const (
+	// Colors for bash highlighting.
+	reset  = "\033[0m"
+	red    = "\033[31m"
+	green  = "\033[32m"
+	italic = "\033[3m"
+	bold   = "\033[1m"
+
+	// OVN related.
+	K8sMgmtIntfName   = "ovn-k8s-mp0"
+	GatewayModeLocal  = "local"
+	GatewayModeShared = "shared"
+)
+
+var (
+	level klog.Level
+)
 
 type SvcInfo struct {
-	IP           string
-	PodName      string
-	PodNamespace string
-	PodIP        string
-	PodPort      string
+	SvcName      string // The service's name
+	SvcNamespace string // The service's namespace
+	ClusterIP    string // The service's cluster IP address
+	PodName      string // The name of one of the pods behind this service
+	PodNamespace string // The namespace of the selected pod
+	PodIP        string // The IP address of the selected pod
+	PodPort      string // One of the pods that the service uses to reach the pod
+}
+
+type NodeInfo struct {
+	NodeExternalBridgeName string // The name of the node's bridge, e.g. breth0 or brex
+	OvnK8sMp0PortName      string // ovn-k8s-mp0
+	OvnK8sMp0OfportNum     string // ofport num of ovn-k8s-mp0
+	K8sNodeNamePort        string // k8s-<nodeName>, e.g. k8s-ovn-worker, only useful for host networked pods
+	NodeName               string // The name of the node that the pod runs on
+	OvnKubePodName         string // The OvnKube pod on the same node as this pod
+	GatewayMode            string // The gateway mode, 'local' or 'shared'
 }
 
 type PodInfo struct {
-	IP                      string
-	MAC                     string
-	VethName                string
-	PortNum                 string
-	LocalNum                string
-	OVNName                 string
-	PodName                 string
-	ContainerName           string
-	OvnKubeContainerPodName string
-	NodeName                string
-	RtosMAC                 string
-	HostNetwork             bool
+	NodeInfo
+	PrimaryInterfaceName string // primary pod interface name inside the pod
```

Comment:
It can be for host networked pods:
podInfo.PrimaryInterfaceName = util.GetLegacyK8sMgmtIntfName(podInfo.NodeName)

for all other pods:
podInfo.PrimaryInterfaceName = "eth0"